### PR TITLE
Add Get_Workflow_Metadata.ipynb

### PR DESCRIPTION
**Pull request by wnojopra**
Add Get_Workflow_Metadata notebook to terra-tools. This notebook offers users a simple option to fetch all or some of the workflow metadata, view it, and optionally export it to BigQuery.

### CODE REVIEW

File: workspace/workspace_description.md [hunk 1]
```diff
@@ -108,11 +108,44 @@ The Python code allows the user to run the code in "dry-run" mode allowing them
 
 -----
 
+## Jupyter Notebook: Get_Workflow_Metadata
+#### **Disclaimer:** **This Notebook is to be implemented at the user’s discretion. We are not responsible for any unexpected behavior (user error or otherwise). Please ensure that you have saved the your table .tsv files and/or data you would like to persist to a permanent location before running this Notebook.**
+
+### **What is this Notebook?**
+When launching large numbers of submissions, it can be difficult to sort through the metadata of individual workflows. For example, agreegating the costs of several hundred workflows requires clicking through each submission. This notebook offers users a simple option to fetch all or some of the workflow metadata, view it, and optionally export it to BigQuery.
```
Comment (mmorgantaylor) [inline workspace/workspace_description.md]: typo "agreegating" -> "aggregating"

File: workspace/workspace_description.md [hunk 2]
```diff
@@ -108,11 +108,44 @@ The Python code allows the user to run the code in "dry-run" mode allowing them
...
+
+
+
+
+### **Notes and Considerations**
+
+If your submission has several workflows, the notebook may take several minutes to run. This is because we must make one API call per workflow. For example, if your submission has 427 workflows, the notebook will make 427 sequential API calls which may take 4-6 minutes.
+
+### **To run this notebook**
+1. Copy this Notebook into the workspace where you want to update the Data Model DRS uris.
```
Comment (mmorgantaylor) [inline workspace/workspace_description.md]: update this line to be specific to this notebook

File: notebooks/Get_Workflow_Metadata.ipynb [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+    "fapi.__version__"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import firecloud.api as fapi"
```
Comment (mmorgantaylor) [inline notebooks/Get_Workflow_Metadata.ipynb]: why is there this import here and in the next cell?

File: notebooks/Get_Workflow_Metadata.ipynb [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+    "If your submission has several workflows, the **notebook may take several minutes to run**. This is because we must make one API call per workflow. For example, if your submission has 427 workflows, the notebook will make 427 sequential API calls which may take 4-6 minutes. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fapi.__version__"
```
Comment (mmorgantaylor) [inline notebooks/Get_Workflow_Metadata.ipynb]: why is this cell here, before importing firecloud.api ? what is the user supposed to do with it?

File: notebooks/Get_Workflow_Metadata.ipynb [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "r = fapi.list_submissions(namespace, workspace)\n",
+    "df = ''\n",
```
Comment (mmorgantaylor) [inline notebooks/Get_Workflow_Metadata.ipynb:65]: I'd move this assigment to the `else` block below - it's a bit confusing here

**Review by mmorgantaylor — APPROVED** [review]
this is great! a few suggestions, mostly cleanup/clarification.

File: notebooks/Get_Workflow_Metadata.ipynb (hunk 3 above)
Comment (wnojopra) [reply notebooks/Get_Workflow_Metadata.ipynb]: These were accidentally left in from debugging. Thanks for catching!

File: notebooks/Get_Workflow_Metadata.ipynb (hunk 5 above)
Comment (wnojopra) [reply notebooks/Get_Workflow_Metadata.ipynb:65]: Ah - technically your suggestion would work as Python variables are scoped to the whole function, not just in the innermost "block" (see https://www.studytonight.com/python-howtos/scope-of-the-variable-initialised-in-the-if-statement). However, given that the df is used outside of the if/else block, it's a good practice to "declare it" outside of the block. This makes it safer from future code changes that modify one branch of the if/else.

Comment (mmorgantaylor) [reply notebooks/Get_Workflow_Metadata.ipynb:65]: hm, that's fair. 👍

**Pull request merged**